src/main/java/ch/njol/skript/expressions/ExprIP.java

```diff
@@ -22,24 +22,33 @@
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 
+import org.bukkit.World;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerLoginEvent;
 import org.eclipse.jdt.annotation.Nullable;
 
+import ch.njol.skript.ScriptLoader;
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Getter;
+import ch.njol.skript.util.Time;
+import ch.njol.util.Kleenean;
 
 @Name("IP")
 @Description("The IP address of player(s).")
 @Examples({"IP-ban the player # is equal to the next line",
 		"ban the IP-address of the player",
 		"broadcast \"Banned the IP %IP of player%\""})
-@Since("1.4, 2.2-dev35")
-public class ExprIP extends SimplePropertyExpression<Player, String> {
+@Since("1.4, 2.2-dev35 (Converted to PropertyExpression)")

```

Comment:
Don't mention implementation details in documentation, the documentation is mainly for scripters and this doesn't add anything useful for them. You're better off not mentioning anything at all if it didn't change the functionality.

Reply:
FFS!!!! Bensku want's me to add it where as others don't. I can't keep switching back and forth. Make a decision and stick to it.

Reply:
You could always write something like `internal restructuring`. That states the change, without confusing skripters at the same time.

Reply:
That's pretty confusing imo, better off how it was originally.

Reply:
Um, hello ? Do you mind explaining how saying `internal change` or the like is less confusing than `changed to PropertyExpression` **for newbies** ??

Reply:
They don't need to know anything, forget it. Like I said before, if anyone cares what was changed in the past, they can look at git.

Reply:
Do, not, mention, about, this, change, in, Since.

Reply:
Just do it how every other file in Skript does it, the version string of when the file was added otherwise `INSERT VERSION` if it’s a new addition.

Reply:
@bensku also said that you shouldn't mention it. Just remove it from the since, it isn't useful for the scripters.

Reply:
I do not want to add it here. If I said otherwise, it must have been a typo or something.